pennylane/beta/tapes/qnode.py

```diff
@@ -0,0 +1,541 @@
+
+        if not provides_jacobian:
+            raise QuantumFunctionError(
+                f"The {device.short_name} device does not provide a native "
+                "method for computing the jacobian."
+            )
+
+        return QuantumTape, interface, "device"
+
+    @staticmethod
+    def _get_parameter_shift_method(device, interface):
+        """Validates whether a particular device and QuantumTape interface
+        supports the parameter-shift differentiation method, and returns
+        the correct tape.
+
+        Args:
+            device (.Device): PennyLane device
+            interface (str): name of the requested interface
+
+        Returns:
+            tuple[.QuantumTape, str, str]: tuple containing the compatible
+            QuantumTape, the interface to apply, and the method argument
+            to pass to the ``QuantumTape.jacobian`` method.
+
+        Raises:
+            QuantumFunctionError: if the device model does not have a corresponding
+            parameter-shift rule.
+        """
+        # determine if the device provides its own jacobian method
+        model = device.capabilities().get("model", None)
+
+        if model == "qubit":
+            # return QubitParamShiftTape, interface, "analytic"
+            warnings.warn(
+                "Qubit parameter-shift rule not yet implemented! Falling back "
+                "to finite-differences."
+            )
+
+            return QuantumTape, interface, "numeric"
+
+        if model == "cv":
+            # return CVParamShiftTape, interface, "analytic"
+            warnings.warn(
+                "CV parameter-shift rule not yet implemented! Falling back "
+                "to finite-differences."
+            )
+
+            return QuantumTape, interface, "numeric"
+
+        raise QuantumFunctionError(
+            f"Device {device.short_name} has model {model} "
+            "that does not support the parameter-shift rule."
+        )
+
+    def construct(self, args, kwargs):
+        """Call the quantum function with a tape context,
+        ensuring the operations get queued."""
+        self.qtape = self._tape()
+
+        # apply the interface (if any)
+        if self.interface is not None:
+            self.INTERFACE_MAP[self.interface](self)
+
+        with self.qtape:
+            measurements = self.func(*args, **kwargs)
+
+        if not isinstance(measurements, Sequence):
+            measurements = (measurements,)
+
+        if not all(isinstance(m, MeasurementProcess) for m in measurements):
+            raise QuantumFunctionError(
+                "A quantum function must return either a single measured observable "
+                "or a nonempty sequence of measured observables."
+            )
+
+        if not all(ret == m for ret, m in zip(measurements, self.qtape.measurements)):
+            raise QuantumFunctionError(
+                "All measurements must be returned in the order they are measured."
+            )

```

Comment:
This is just a consistency check to make sure that the return statement matches the order of the measurements in the queue